compliance/virtualmachines/agent/vsock/client_test.go

```diff
@@ -0,0 +1,82 @@
+package vsock
+
+import (
+	"fmt"
+	"net"
+	"testing"
+	"time"
+
+	v4 "github.com/stackrox/rox/generated/internalapi/scanner/v4"
+	v1 "github.com/stackrox/rox/generated/internalapi/virtualmachine/v1"
+	"github.com/stackrox/rox/pkg/protoassert"
+	"github.com/stackrox/rox/pkg/testutils"
+	"github.com/stackrox/rox/pkg/utils"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/protobuf/proto"
+)
+
+func setupLocalTCPListener(t *testing.T) (net.Listener, uint64) {
+	port := testutils.GetFreeTestPort()
+	listener, err := net.Listen("tcp", fmt.Sprintf("127.0.0.1:%d", port))
+	require.NoError(t, err, "failed to create local TCP listener")
+
+	t.Cleanup(func() {
+		utils.IgnoreError(listener.Close)
+	})
+
+	return listener, port
+}
+
+func TestClient_writeIndexReport_LocalSocket(t *testing.T) {
+	listener, port := setupLocalTCPListener(t)
+	client := &Client{Port: uint32(port)}
+	testReport := &v1.IndexReport{
+		IndexV4: &v4.IndexReport{
+			HashId:  "test-hash-local",
+			State:   "completed",
+			Success: true,
+		},
+	}
+
+	receivedData := make(chan []byte, 1)
+	listenerErr := make(chan error, 1)
```

Comment:
Style: The first one is being closed only on timeout or when an error happens. I think, we can safely close it unconditionally in a deferred statement.
`listenerErr` is not being closed at all.